File: TenMinsPy/day74.py
# Ask for a password input, if new, require strong password and register, if password exists(hashed password txt file), require existing password

import os
import hashlib

class day74:
    def checkPass(s):
        # checks the lenght of password
        if len(s) >= 8:
            passScore = 0
            usedLetters = []
            usedSigns = []
            # does not take space
            for c in s:
                if c.isspace():
                    return "error there is space"
                # check if letter is repeated and for symbols and numbers
                elif c.isalpha():
                    c = c.upper()
                    if c not in usedLetters:
                        usedLetters.append(c)
                    else:
                        passScore -= 5
                else:
                    if c not in usedSigns:
                        usedSigns.append(c)
                    else:
                        passScore -= 5
            # checks ratio of upper to lower characters
            upper = 0
            lower = 0
            for c in s:
                if c.isupper():
                    upper += 1
                if c.islower():
                    lower += 1
            # checks password score from given parameters
            for i in range((len(usedLetters) + len(usedSigns))):
                passScore += 3
            # returns values based on password score and other ideas
            if passScore <= 10 or len(usedSigns) <= 2 or abs(upper - lower) > 2:
                return "weak"
            elif passScore >= 11 and passScore <= 19 or len(usedSigns) <= 3 or abs(upper - lower) < 2:
                return "moderate"
            elif passScore >= 20 or len(usedSigns) <= 4:
                return "strong"
            else:
                return "error"
        else:
            return "Enter longer password"
    def checkExist(s):
        if day74.checkPass(s) == "strong" or day74.checkPass(s) == "moderate":
            path = "D:/coding30minsAday/textFiles/textDay74P.txt"
            if os.path.exists(path):
                s = input("Enter password\n")
                newS = hashlib.sha1(s.encode()).hexdigest()
                with open(path, "r") as f:
                    if newS in f.read():
                        print("nice")
                    else:
                        print("missing")
            else:
                s = input("enter new password\n")
                newP = hashlib.sha1(s.encode()).hexdigest()
                with open(path, "w") as f:
                    f.write(newP)

s = input("Enter password\n")
day74.checkExist(s)

File: TenMinsJava/day74.java
// Ask for a password input, if new, require strong password and register, if password exists(hashed password txt file), require existing password
package TenMinsJava;

import java.io.File;
import java.io.FileWriter;
import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.ArrayList;
import java.util.Scanner;

public class day74 {
    public static String checkPass(String s) {
        String result;
        if (s.length() >= 8) {
            int passScore = 0;
            ArrayList<Character> usedLetters = new ArrayList<>();
            ArrayList<Character> usedSigns = new ArrayList<>();
            for (Character c : s.toCharArray()) {
                if (Character.isWhitespace(c)) {
                    result = "error";
                }
                else {
                    if (Character.isAlphabetic(c)) {
                        c = Character.toUpperCase(c);
                        if (!usedLetters.contains(c)) {
                            usedLetters.add(c);
                        }
                        else {
                            passScore -= 5;
                        }
                    }
                    else {
                        if (!usedLetters.contains(c)) {
                            usedSigns.add(c);
                        }
                        else {
                            passScore -= 5;
                        }
                    }
                }
            }
            // pwede ilagay sa taas
            int upper = 0;
            int lower = 0;
            for (Character c : s.toCharArray()) {
                if (Character.isUpperCase(c)) {
                    upper++;
                }
                if (Character.isLowerCase(c)) {
                    lower++;
                }
            }
            for (int i = 0; i < usedLetters.size() + usedSigns.size(); i++) {
                passScore += 3;
            }
            if (passScore <= 10 || usedSigns.size() <= 2 || Math.abs(upper - lower) > 2) {
                result = "weak";
            } 
            else if (passScore >= 11 && passScore <= 19 || usedSigns.size() <= 3 || Math.abs(upper - lower) < 2) {
                result = "moderate";
            } 
            else if (passScore >= 20 || usedSigns.size() <= 4) {
                result = "strong";
            } 
            else {
                result = "error";
            }
        }
        else {
            result = "error";
        }
        return result;
    }
    public static String getSha1(String s) {
        try {
            MessageDigest sha = MessageDigest.getInstance("SHA1");
            byte[] b = sha.digest(s.getBytes());
            BigInteger n = new BigInteger(1, b);
            String newP = n.toString(16);
            while (newP.length() < 40) {
                newP = "0" + newP;
            }
            return newP;
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
    public static void main(String[] args) {
        try {
            System.out.println("hello");
            File f = new File("D:/coding30minsAday/textFiles/textDay74J.txt");
            if (f.exists()) {
                Scanner scan = new Scanner(System.in);
                System.out.println("Enter password");
                String s = scan.nextLine();
                if (checkPass(s).equals("strong") || checkPass(s).equals("moderate")) {
                    Scanner fScan = new Scanner(f);
                    while (fScan.hasNextLine()) {
                            if (fScan.nextLine().trim().equals(getSha1(s))) {
                                System.out.println("nice");
                                break;
                            }
                            else {
                                System.out.println("missing");
                                break;
                            }
                    }
                    fScan.close();
                }
                scan.close();
            }
            else {
                Scanner scan = new Scanner(System.in);
                System.out.println("Enter password");
                String s = scan.nextLine();
                FileWriter w = new FileWriter("D:/coding30minsAday/textFiles/textDay74J.txt");
                w.write(getSha1(s));
                w.close();
                scan.close();
            }
        } catch (Exception e) {
            System.err.println("error");
        }
    }
}